File: ios/Classes/LocationData.swift
//
//  LocationData.swift
//  flutter_location_manager
//
//  Created by Fellype Macedo on 18/07/19.
//

import Foundation
import CoreLocation

struct Location : Codable{
    
    let latitude : Double
    let longitude: Double
    let timestamp: Double
    let speed: Double
    let accuracy: Double
    let altitude: Double
    
    init(location: CLLocation){
        self.latitude = location.coordinate.latitude
        self.longitude = location.coordinate.longitude
        self.timestamp = location.timestamp.timeIntervalSince1970
        self.speed = location.speed
        self.accuracy = location.horizontalAccuracy
        self.altitude = location.altitude
    }
    
    func toJson() -> String{
        let jsonEncoder = JSONEncoder()
        let jsonData = try? jsonEncoder.encode(self)
        return String(data: jsonData!, encoding: String.Encoding.utf8) ?? ""
    }
}

File: android/src/main/kotlin/com/layunne/flutter_location_manager/models/LocationData.kt
package com.layunne.flutter_location_manager.models

import android.location.Location
import org.json.JSONObject

data class LocationData(
        val latitude: Double,
        val longitude: Double,
        val timestamp: Long,
        val speed: Float,
        val accuracy: Float,
        val altitude: Double
) {
    constructor(location: Location):
            this(
                    location.latitude,
                    location.longitude,
                    location.time,
                    location.speed,
                    location.accuracy,
                    location.altitude
            )

    fun toJson() : String {
        val json = JSONObject()
        json.put("latitude", this.latitude)
        json.put("longitude", this.longitude)
        json.put("timestamp", this.timestamp)
        json.put("accuracy", this.accuracy)
        json.put("altitude", this.altitude)
        json.put("speed", this.speed)

        return json.toString()
    }
}